main.py

```diff
@@ -0,0 +1,215 @@
+"""
+    To add categories print 1
+    If you want to see the list print 2
+                    Group by:
+    To group by month you need to print 3
+    To group by category you need to print 4
+    If you want to group by month and by category print 5
+    To show your expenses as a chart print 6
+    To create a file with your expenses print 7
+    To stop interacting with data print 8
+"""
+
+import csv
+from datetime import datetime
+import re
+import matplotlib.pyplot as plt
+
+fieldnames = ["Category", "Date", "Amount"]
+categories = {}
+
+
+def check_int(prompt):
+    """
+    Checking the correctness of the number input
+    """
+    while True:
+        try:
+            enter_value = float(input(prompt))
+        except ValueError:
+            print("Try again!")
+        if enter_value <= 0:
+            print("Try again!")
+        else:
+            break
+    return enter_value
+
+
+def menu():
+    print(
+        """
+    1. Add a categories
+    2. Show a list
+    3. Group by month
+    4. Group by category
+    5. Group by month and category
+    6. View diagram
+    7. Create file
+    8. Exit
+    """
+    )
+    choice_ = check_int("Enter your choice: ")
+    return int(choice_)
+
+
+def check_string(prompt):
+    """
+    Checking the correctness of the string input
+    """
+    while True:
+        try:
+            enter_str = input(prompt)
+        except ValueError:
+            print("Try again!")
+        if re.match("^[0-9]*$", enter_str):
```

Comment:
`if !re.match: break` is better. I won't need to use else

Reply:
it will be fixed